### ISSUE TITLE
Large integers are truncated

### ISSUE THREAD

**Issue by rossviljoen** [JuliaPluto/PlutoPlotly.jl#61]
Large `Int32`s in the plot data seem to get truncated, resulting in an unwanted loss of precision when plotting with `PlutoPlot(Plot(...))`, this is fixed when setting `PlutoPlotly.floatval() = Val(false)`, so I think it's related to the preprocessing.

**Comment by disberd** [comment]
Indeed by default all numbers are converted to Float32 for saving memory and speeding up transfer between JS and julia.
You can control this with `PlutoPlotly.FORCE_FLOAT32[] = false` but I guess overwriting the method achieves the same result.

I can probably skip converting Int32s into Float32 but not sure whether converting Int64 to Int32 or Float32. Any opinion on this @rossviljoen?

**Comment by rossviljoen** [comment]
I don't believe `PlutoPlotly.FORCE_FLOAT32[] = false` will work with a `ScopedValue`, as there's no `setindex!` method for it, hence why it's necessary to override the method. I'm not sure there's that much benefit to it being a ScopedValue anyway given the issue with show you mentioned.

> I can probably skip converting Int32s into Float32 but not sure whether converting Int64 to Int32 or Float32

If it's necessary for performance, converting `Int64` to `Float32` likely makes more sense to avoid overflow issues. I think it would be good to document this behaviour (and how to disable it) in general though.

(PS. thanks for the excellent package!)
